apps/wing-console/console/app/demo/main.w

```diff
@@ -1,13 +1,64 @@
 bring cloud;
 bring ex;
+bring ui;
+
 // @see https://github.com/winglang/wing/issues/4237 it crashes the Console preview env.
 //let secret = new cloud.Secret(name: "my-secret");
 
 let bucket = new cloud.Bucket();
 let queue = new cloud.Queue();
 let api = new cloud.Api();
 let counter = new cloud.Counter(initial: 0);
+struct PutData {
+  fileName: str;
+  fileContent: str;
+}
+struct GetData {
+  fileName: str;
+}
+struct DeleteData {
+  fileName: str;
+}
+
+class myBucket {
+  b: cloud.Bucket;
+  new() {
+    this.b = new cloud.Bucket();
+    new ui.FileBrowser("File Browser",
+      {
+        put: inflight (payload: str) => {
+          let data: PutData = PutData.fromJson(Json.parse(payload));
```

Comment:
is there a way to avoid it and pass a struct instead of a str?

Reply:
there is also this one:
https://github.com/winglang/wing/pull/6219

Reply:
@skyrpex but the implementation of the handlers is done in the custom resource `.w` file